File: sanidad-backend/src/main/java/com/farmacia/sanidadbackend/dto/LoteDetalleRequest.java
package com.farmacia.sanidadbackend.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Data;

@Data
public class LoteDetalleRequest {

    @NotNull(message = "El medicamento es obligatorio")
    private Long medicamentoId;

    @NotNull(message = "La cantidad es obligatoria")
    @Positive(message = "La cantidad debe ser mayor que cero")
    private Integer cantidad;
}

File: SanidadMovil/app/src/main/java/com/sanidad/movil/data/remote/dto/LoteDetalleRequest.kt
package com.sanidad.movil.data.remote.dto

data class LoteDetalleRequest(
    val medicamentoId: Long,
    val cantidad: Int,
    val rackId: Long? = null,
    val nivel: Int? = null,
    val columna: Int? = null,
    val profundidadIndex: Int? = null
)